### DISCUSSION TITLE
Pull request update branch doesn't respect repo merge settings

### DISCUSSION THREAD

**Discussion by IliaFeldgun** [community/community#50521] [Pull Requests]
### Select Topic Area

Bug

### Body

Consider this scenario:
1. A pull request with a fresh commit was opened, 1 commit after main branch.
2. Someone else added a commit to the main branch, thus the pull request needs to be either rebased or merged.
![Screenshot from 2023-03-20 19-47-44](https://user-images.githubusercontent.com/41454303/226425922-2288eb59-4db0-4a4e-9be7-b060e2fcc2ef.png)
3. The repository is configured to use only the rebase method, but this setting doesn't include preventing merge within other branches, nor there seems to be any other setting controlling the feature.
![Screenshot from 2023-03-20 19-47-19](https://user-images.githubusercontent.com/41454303/226426238-36cb6eb1-2eaf-40d7-9ae0-5400880efe26.png)

We would like to prevent ourselves from being able to `Update Branch` with a merge commit. The reasoning behind this:
1. We consider rebase to be the right way to align our orgs branches.
2. The merge commit is the default option, and clicking on it is too easy. And we basically have to remove it from our history either way.
3. If a merge commit was accidentally used, we need to use rebase+fixup.
4. Rebasing in local git would mean the PR review would become stale, unlike with Githubs' rebase, which leaves the review status.

Should I just go and open a support ticket, or am I missing something?

**Comment by IliaFeldgun** [comment]
I think the "Update Branch" screen is available only if you require the main branch to be "up to date".
![Screenshot from 2023-03-20 20-36-01](https://user-images.githubusercontent.com/41454303/226435000-7371110d-ed79-4f97-b532-992bd7dcf91c.png)
Note that the "only rebase" setting is repository-wide.

**Comment by IliaFeldgun** [comment]
I also found a hack for now, to reverse the behavior.
1. Go back to your local repo
2. `git fetch`
3. Make sure you're in the right branch
4. `git push --force-with-lease`
5. This will allow you to retry the "Rebase branch" option, and not dismiss the review after the rebase.
![Screenshot from 2023-03-20 20-43-07](https://user-images.githubusercontent.com/41454303/226436512-e24cef3e-de46-451c-99c7-536d5b1c9615.png)

Of course I'm not considering conflicts within this scope, as they are handled differently, and would require re-review either way.

**Comment by yuvalta** [comment]
Yeah it's a problem for me too

**Comment by TheGittyPerson** [comment]
Also a problem for me
